File: src/frontend/src/types/dictionaryService/DictionaryDto.ts
import {Phonetic} from "@/types/dictionaryService/Phonetic";
import {Meaning} from "@/types/dictionaryService/Meaning";
import {License} from "@/types/dictionaryService/License";

export interface DictionaryDto {
    word: string;
    phonetics: Phonetic[];
    meanings: Meaning[];
    license: License;
    sourceUrls: string[];
}

File: src/main/java/org/williard/VueAndJava/dto/DictionaryDto.java
package org.williard.VueAndJava.dto;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;

@Data
@NoArgsConstructor
public class DictionaryDto {

    public String word;
    public ArrayList<Phonetic> phonetics;
    public ArrayList<Meaning> meanings;
    public License license;
    public ArrayList<String> sourceUrls;

}
